packages/timeline/src/browser/timeline-contribution.ts

```diff
@@ -0,0 +1,108 @@
+/********************************************************************************
+ * Copyright (C) 2020 RedHat and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, inject } from 'inversify';
+import {
+    FrontendApplicationContribution,
+    FrontendApplication,
+    ViewContainer,
+    WidgetManager,
+    Widget,
+    ApplicationShell,
+    Navigatable
+} from '@theia/core/lib/browser';
+import { FileNavigatorContribution } from '@theia/navigator/lib/browser/navigator-contribution';
+import { EXPLORER_VIEW_CONTAINER_ID } from '@theia/navigator/lib/browser';
+import { TimelineWidget } from './timeline-widget';
+import { TimelineService } from './timeline-service';
+import { Command, CommandRegistry } from '@theia/core/lib/common';
+import { TabBarToolbarRegistry } from '@theia/core/lib/browser/shell/tab-bar-toolbar';
+
+@injectable()
+export class TimelineContribution implements FrontendApplicationContribution {
+
+    @inject(FileNavigatorContribution)
+    protected readonly explorer: FileNavigatorContribution;
+    @inject(WidgetManager)
+    protected readonly widgetManager: WidgetManager;
+    @inject(TimelineService)
+    protected readonly timelineService: TimelineService;
+    @inject(CommandRegistry)
+    protected readonly commandRegistry: CommandRegistry;
+    @inject(TabBarToolbarRegistry)
+    protected readonly tabBarToolbar: TabBarToolbarRegistry;
+    @inject(ApplicationShell)
+    protected readonly shell: ApplicationShell;
+
+    public static readonly LOAD_MORE_COMMAND: Command = {
+        id: 'timeline-load-more'
+    };
+
+    async onDidInitializeLayout?(app: FrontendApplication): Promise<void> {
+        const explorer = await this.widgetManager.getWidget(EXPLORER_VIEW_CONTAINER_ID);
+        let timeline: TimelineWidget;
+        this.timelineService.onDidChangeProviders( async event => {
+            if (explorer instanceof ViewContainer) {
```

Comment:
we should not capture explorer, but retrieve it from `WidgetManager` always

Reply:
done